### DISCUSSION TITLE
Welcome to scenery Discussions!

### DISCUSSION THREAD

**Discussion by skalarproduktraum** [scenerygraphics/scenery#354] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by jimbok8** [comment]
This looks really interesting - I am about to dive in and have a go!
I am looking to render large point clouds, I guess this is as good (or better?) than anything I can do in Jogl (Jogamp) either directly or using a scenegraph such as Ardor3D (Jogamp Continuation)?

I am keen to integrate with JavaFX - I see this has been moved, is it still supported? Are there performance penalties in using JavaFX? 
Has anybody tried the new NewtCanvasJFX?

**Reply by skalarproduktraum** [reply]
Hey @jimbok8 and welcome!

With rendering large point clouds, you've come to use at kind of the right time, we wanted to look deeper into this now because we have more users from localisation microscopy now, which generates a point clouds. In contrast to Jogl or Ardor, we not only support OpenGL as rendering backend, but also Vulkan (and actually prefer that, we might phase out OpenGL at some point in the mid-term future). 

Integration with JavaFX is unfortunately quite hard, because JavaFX keeps the parts that access the GPU quite closed, so rendering with to a JavaFX window usually incurs a quite massive performance penalty because of buffer copies. For Swing/AWT we solve that by using lwjgl-awt (for Vulkan) or JOGL's NewtAWTCanvas. We have been following [the DriftFX project](https://github.com/eclipse-efx/efxclipse-drift/) but did not yet have a chance to explore it really. You are most welcome to check it out, and we are very open to contributions 😎 

And of course you are also very welcome to join our developer chat at https://gitter.im/scenerygraphics/SciView!

cheers!

**Reply by skalarproduktraum** [reply]
PS: For getting started, we also have some docs available at https://docs.scenery.graphics/, they are a bit preliminary, but might help you get started 👍

**Reply by jimbok8** [reply]
I see you have a compute shader example. With reference to point clouds I have just seen this:
"Rendering Point Clouds with Compute Shaders" by Markus Schutz and Michael Wimmer (TU Wien).
They say:
"We propose a compute shader based point cloud rasterizer with up to 10 times higher performance than classic point-based rendering with the GL_POINT primitive. In addition to that, our rasterizer offers 5 byte depth-buffer precision with uniform or customizable distribution, and we show that it is possible to implement a high-quality splatting method that blends together overlapping fragments while still maintaining higher frame-rates than the traditional approach."

Might this be a way forward, or are you doing this already?
Jim

**Reply by skalarproduktraum** [reply]
We are actually still doing the very, very cheap GL_POINT version... *hides in shame*

Would you like to help us out by implementing that? We'd be happy to lend you a hand either here or on the gitter at gitter.im/scenerygraphics/sciview 👍
